src/net/ssl-context.cpp

```diff
@@ -39,14 +41,21 @@ void SslContext::init(std::string ca_bundle_path) {
         throw std::exception();
     }
 
-    if (SSL_CTX_load_verify_locations(ctx, ca_bundle_path.c_str(), NULL) == 0) {
+    if (SSL_CTX_load_verify_locations(ctx, ca_bundle_path.c_str(), NULL) != 1) {
         debug("ssl: failed to load verify location");
         throw std::exception();
     };
 
     SSL_CTX_set_verify(ctx, SSL_VERIFY_PEER, NULL);
 }
 
+Var<SslContext> SslContext::global() {
+    if (global_ == nullptr){
+        global_ = new SslContext();
+    }
+    return Var<SslContext>(global_);
```

Comment:
This is not correct: calling `Var<SslContext>(global_)` you create a smart point with reference counting equal to one and associated pointer `global_`. The returned `Var<>` can be copied but, when it and all its copies go out of scope, this leads to `delete global_` being called. I suggest instead this pattern for the singleton:

```
Var<SslContext> SslContext::global() {
    static Var<SslContext> singleton(new SslContext);
    return singleton;
}```

This works because, since `singleton` is static, it is initialized only once the first time that `global()` is called.